Makefile

```diff
@@ -501,6 +501,7 @@ fmt: tools ## Run goimports
 vet: ## Run go vet
 	$(GO) vet ./...
 
+# Run misspell check and various lints
```

Comment:
@stmcginnis : Makefile targets `check` and `lint` seems repetitive, we can just keep `lint` and remove `check`. WDYT ?

Reply:
Yeah, that seems cleaner to me. Especially since we are running other targets as part of `lint`, there isn't much value to have `check` just to add one more target to the list.

Reply:
updated, PTAL